Panels/tests/Header-specs.js

```diff
@@ -103,105 +88,53 @@ describe('Header Specs', () => {
 			</Header>
 		);
 
-		const actual = subject.find(`.${css.slotAfter}`).first().text();
+		const slotAfter = screen.getByText(expected).parentElement;
 
-		expect(actual).toBe(expected);
+		expect(slotAfter).toHaveClass(css.slotAfter);
 	});
 
 	test('should not render back button', () => {
-		const subject = mount(
-			<Header />
-		);
+		render(<Header />);
 
-		const backButton = subject.find(`.${css.slotBefore}`).find('Button');
-		const expected = 0;
-		const actual = backButton.length;
+		const backButton = screen.queryByLabelText('go to previous');
 
-		expect(actual).toBe(expected);
+		expect(backButton).toBeNull();
 	});
 
 	test('should render close button when \'noCloseButton\' is not specified', () => {
-		const subject = mount(
-			<Header />
-		);
+		render(<Header />);
 
-		const closeButton = subject.find(`.${css.slotAfter}`).find('Button');
-		const expected = 1;
-		const actual = closeButton.length;
+		const closeButton = screen.getByLabelText('Exit app');
 
-		expect(actual).toBe(expected);
+		expect(closeButton).toBeInTheDocument();
 	});
 
 	test('should not render close button when \'noCloseButton\' is set to true', () => {
-		const subject = mount(
-			<Header noCloseButton />
-		);
+		render(<Header noCloseButton />);
 
-		const closeButton = subject.find(`.${css.slotAfter}`).find('Button');
-		const expected = 0;
-		const actual = closeButton.length;
+		const closeButton = screen.queryByLabelText('Exit app');
 
-		expect(actual).toBe(expected);
+		expect(closeButton).toBeNull();
 	});
 
 	test('should call onClose when close button is clicked', () => {
```

Comment:
You can use async to get rid of the console warning.
```suggestion
	test('should call onClose when close button is clicked', async () => {
		const handleClose = jest.fn();
		render(<Header onClose={handleClose} />);

		userEvent.click(screen.getByRole('button'));

		await waitFor(() =>
			expect(handleClose).toHaveBeenCalled()
		)
        });
```